shared/pkg/scanner/trivy/local.go

```diff
@@ -0,0 +1,256 @@
+	a.logger.Infof("Called %s scanner on source %v %v", ScannerName, sourceType, userInput)
+	go func() {
+		var output bytes.Buffer
+
+		dbRepoDefaultValue, ok := trivyFlag.DBRepositoryFlag.Value.(string)
+		if !ok {
+			a.setError(fmt.Errorf("unable to get trivy DB repo config"))
+			return
+		}
+
+		severities := []trivyDBTypes.Severity{}
+		for _, name := range trivyDBTypes.SeverityNames {
+			sev, err := trivyDBTypes.NewSeverity(strings.ToUpper(name))
+			if err != nil {
+				a.setError(fmt.Errorf("unable to get trivy severities: %w", err))
+				return
+			}
+			severities = append(severities, sev)
+		}
+
+		trivyOptions := trivyFlag.Options{
+			GlobalOptions: trivyFlag.GlobalOptions{
+				Timeout: a.config.Timeout,
+			},
+			ScanOptions: trivyFlag.ScanOptions{
+				Target: userInput,
+				SecurityChecks: []string{
+					trivyTypes.SecurityCheckVulnerability, // Enable just vuln scanning
+				},
+			},
+			ReportOptions: trivyFlag.ReportOptions{
+				Format:       "json",  // Trivy's own json format is the most complete for vuls
+				ReportFormat: "all",   // Full report not just summary
+				Output:       &output, // Save the output to our local buffer instead of Stdout
+				ListAllPkgs:  false,   // Only include packages with vulnerabilities
+				Severities:   severities,
+			},
+			DBOptions: trivyFlag.DBOptions{
+				DBRepository: dbRepoDefaultValue, // Use the default trivy source for the vuln DB
+				NoProgress:   true,               // Disable the interactive progress bar
+			},
+			VulnerabilityOptions: trivyFlag.VulnerabilityOptions{
+				VulnType: []string{trivyTypes.VulnTypeOS, trivyTypes.VulnTypeLibrary},
+			},
+		}
+
+		var trivySourceType artifact.TargetKind
+		switch sourceType {
+		case utils.IMAGE:
+			trivySourceType = artifact.TargetContainerImage
+		case utils.ROOTFS:
+			trivySourceType = artifact.TargetRootfs
+		case utils.DIR, utils.FILE:
+			trivySourceType = artifact.TargetFilesystem
+		case utils.SBOM:
+			trivySourceType = artifact.TargetSBOM
+		default:
+			a.logger.Infof("Skipping analyze unsupported source type: %s", sourceType)
+			a.resultChan <- &scanner.Results{
+				Matches: nil, // empty results,
+				ScannerInfo: scanner.Info{
+					Name: ScannerName,
+				},
+			}
+			return
+		}
+
+		err := artifact.Run(context.TODO(), trivyOptions, trivySourceType)
+		if err != nil {
+			a.setError(fmt.Errorf("failed to generate SBOM: %w", err))
+			return
+		}
+
+		a.logger.Infof("Sending successful results")
+		a.resultChan <- a.CreateResult(output.Bytes())
+	}()
+
+	return nil
+}
+
+func getCVSSesFromVul(vCvss trivyDBTypes.VendorCVSS) []scanner.CVSS {
+	// TODO(sambetts) Work out how to include all the
+	// CVSS's by vendor ID in our report, for now only
+	// collect one of each version.
+	cvsses := []scanner.CVSS{}
+	v2Collected := false
+	v3Collected := false
+	for _, cvss := range vCvss {
+		if cvss.V3Vector != "" && !v3Collected {
+			cvsses = append(cvsses, scanner.CVSS{
+				Version: "3.1",
```

Comment:
maybe better to extract version from vector like in `getCVSSV3Version` (shared/pkg/scanner/dependency_track/dependency_track.go)